pkg/quota/admission/podlimitrequest/admission_test.go

```diff
@@ -0,0 +1,223 @@
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/auth/user"
+	"k8s.io/kubernetes/pkg/client/cache"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"speter.net/go/exp/math/dec/inf"
+)
+
+const (
+	yamlConfig = `
+apiVersion: v1
+kind: PodLimitRequestConfig
+enabled: true
+limitCPUToMemoryRatio: 1.0
+cpuRequestToLimitRatio: 0.1
+memoryRequestToLimitRatio: 0.25
+`
+	invalidConfig = `
+apiVersion: v1
+kind: PodLimitRequestConfig
+enabled: true
+cpuRequestToLimitRatio: 2.0
+`
+	invalidConfig2 = `Enabled: true`
+)
+
+const (
+	oneGiB  = 1024 * 1024 * 1024
+	oneCore = 1000 // milliCores, scaled by 3
+)
+
+var (
+	configMeta = unversioned.TypeMeta{Kind: "PodLimitRequestConfig", APIVersion: "v1"}
+	//configMeta   = unversioned.TypeMeta{}
+	targetConfig = api.PodLimitRequestConfig{
+		TypeMeta:                  configMeta,
+		Enabled:                   true,
+		LimitCPUToMemoryRatio:     1.0,
+		CPURequestToLimitRatio:    0.1,
+		MemoryRequestToLimitRatio: 0.25,
+	}
+)
+
+func TestConfigReader(t *testing.T) {
+	initial := testConfig(true, 0.1, 0.2, 0.3)
+	if config, err := json.Marshal(initial); err != nil {
+		t.Errorf("json.Marshal: config serialize failed: %v", err)
+	} else if returned, readerr := ReadConfig(bytes.NewReader(config)); readerr != nil {
+		t.Errorf("ReadConfig: config deserialize failed: %v", readerr)
+	} else if returned != initial {
+		t.Errorf("ReadConfig: expected %v, got %v", initial, returned)
+	}
+
+	if _, err := ReadConfig(bytes.NewReader([]byte("asdfasdfasdF"))); err == nil {
+		t.Errorf("should have choked on broken config")
+	}
+	if config, err := ReadConfig(bytes.NewReader([]byte(yamlConfig))); err != nil {
+		t.Errorf("should have been able to deserialize yaml: %v", err)
+	} else if config != targetConfig {
+		t.Errorf("target for yamlConfig: was %v, should have been %v", config, targetConfig)
+	}
+	if config, err := ReadConfig(bytes.NewReader([]byte(invalidConfig))); err != nil {
+		t.Errorf("ReadConfig invalidConfig: config deserialize failed: %v", err)
+	} else if err2 := Validate(config); err2 == nil {
+		t.Errorf("should have choked on out-of-bounds ratio")
+	}
+	if config, err := ReadConfig(bytes.NewReader([]byte(invalidConfig2))); err != nil {
+		t.Errorf("ReadConfig invalidConfig2: config deserialize failed: %v", err)
+	} else if err2 := Validate(config); err2 == nil {
+		t.Errorf("should have complained about no ratios being set")
+	}
+
+}
+
+func TestLimitRequestAdmission(t *testing.T) {
+	tests := []struct {
+		name               string
+		config             api.PodLimitRequestConfig
+		object             runtime.Object
+		expectedMemRequest *inf.Dec
```

Comment:
any reason you cant just use resource.Quantity here?  you can then use quantity.MustParse so its easier to read the test case then having to convert to the inf.Dec structure.